sklearn/linear_model/sag.py

```diff
@@ -0,0 +1,355 @@
+"""Solvers for Ridge and LogisticRegression using SAG algorithm"""
+
+# Authors: Danny Sullivan <dbsullivan23@gmail.com>
+#          Tom Dupre la Tour <tom.dupre-la-tour@m4x.org>
+#
+# Licence: BSD 3 clause
+
+import numpy as np
+import scipy.sparse as sp
+import warnings
+
+from ..utils import ConvergenceWarning
+from ..utils.fixes import astype
+from ..utils.seq_dataset import ArrayDataset, CSRDataset
+from .sgd_fast import Log, SquaredLoss
+from .sag_fast import sag_sparse, get_auto_eta
+
+MAX_INT = np.iinfo(np.int32).max
+
+"""For sparse data intercept updates are scaled by this decay factor to avoid
+intercept oscillation."""
```

Comment:
Is there a reference for this magic number? Maybe mention the paragraph of the paper where this is documented.

Also: please use the `#` sign for inline comments. `"""` should be reserved for docstrings. Is there.